networkparams_test.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2015-2018 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/decred/dcrd/blockchain"
+	"github.com/decred/dcrd/chaincfg"
+)
+
+// checkPowLimitsAreConsistent ensures PowLimit and PowLimitBits are consistent with each other
+// PowLimit:         mainPowLimit,// big int
+// PowLimitBits:     0x1d00ffff,  // conceptually the same
+//                                // value, but in an obscure form
+func checkPowLimitsAreConsistent(t *testing.T, params chaincfg.Params) {
+	powLimitBigInt := params.PowLimit
+	powLimitCompact := params.PowLimitBits
+
+	toBig := blockchain.CompactToBig(powLimitCompact)
+	toCompact := blockchain.BigToCompact(powLimitBigInt)
+
+	// Check params.PowLimitBits matches params.PowLimit converted into the compact form
+	if toCompact != powLimitCompact {
+		// Print debug info
+		fmt.Println("params.PowLimit    :", fmt.Sprintf("%064x", powLimitBigInt))
+		fmt.Println("                   :", fmt.Sprintf("%x", toCompact)) // PoW limit converted to compact
+		fmt.Println("params.PowLimitBits:", fmt.Sprintf("%064x", toBig))  // Bits converted to big.Int
+		fmt.Println("                   :", fmt.Sprintf("%x", powLimitCompact))
+		fmt.Println()
+
+		t.Fatalf("PowLimit values mismatch: "+
+			"params.PowLimit=%v is not consistent with the "+
+			"params.PowLimitBits=%v",
+			fmt.Sprintf("%064x", powLimitBigInt), fmt.Sprintf("%x", powLimitCompact))
```

Comment:
No need to call `fmt.Sprintf` again since `t.Fatalf` already support formatting.

e.g.

```Go
		t.Fatalf("PowLimit values mismatch: params.PowLimit=%064x is "+
			"inconsistent with params.PowLimitBits=%x", powLimitBigInt,
			powLimitCompact)
```